### PR TITLE
Use the image storage pool for volumes and instances

### LINKED ISSUE #307: Use instance affinity policy on ibm_pi_volume
In reference to https://github.com/IBM-Cloud/terraform-provider-ibm/pull/2800 fix on the pi_volume resource, we need to use the instance affinity policy on the volume that are created by the automation.

With this option we don't need the volume_type aka diskType. This will prevent the errors we are getting recently on some regions due to storage pool placement. Also, there is no need of existing logic where a data source fetch the storage type from the image.

Note that this fix will be released in the next iteration of IBM provider plugin.

### PR DESCRIPTION
**Pull request by yussufsh**
Signed-off-by: Yussuf Shaikh <yussuf.shaikh@ibm.com>

Fixes #307 #330

### CODE REVIEW

**Comment by yussufsh** [comment]
Fixes #330 #307

**Review by Prajyot-Parab — DISMISSED** [review]
/lgtm

**Review by aishwaryabk — DISMISSED** [review]
/lgtm

File: modules/1_prepare/prepare.tf [hunk 1]
```diff
@@ -82,12 +82,19 @@ resource "ibm_pi_instance" "bastion" {
...
   pi_health_status     = var.bastion_health_status
   pi_volume_ids        = var.storage_type == "nfs" ? ibm_pi_volume.volume.*.volume_id : null
+  pi_storage_pool      = local.bastion_storage_pool
+
+  pi_network {
+    network_id = ibm_pi_network.public_network.network_id
+  }
+  pi_network {
+    network_id = data.ibm_pi_network.network.id
+  }
```
Comment (mkumatag) [inline modules/1_prepare/prepare.tf:97]: why are we assigning twice here?

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Review by mkumatag — APPROVED** [review]
Please make sure we add proper documentation in the release notes because we are changing the variable names, and also update the description in the PR about the networking changes done.

/lgtm

**Review by Prajyot-Parab — APPROVED** [review]
/approve

**Pull request merged**